### PR TITLE
fix: [CDS-25632]: Handle env Update via yaml

### PR DESCRIPTION
**Pull request by abhinavmittal-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30741)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java [hunk 1]
```diff
@@ -1133,7 +1133,29 @@ public Workflow updateLinkedWorkflow(Workflow workflow, Workflow existingWorkflo
     // Update Workflow Phase steps
     workflowServiceTemplateHelper.updateLinkedWorkflowPhases(
         orchestrationWorkflow.getWorkflowPhases(), existingOrchestrationWorkflow.getWorkflowPhases(), fromYaml);
-    return updateWorkflow(workflow, workflow.getOrchestrationWorkflow(), false);
+    boolean envChanged = false;
+    if (workflow.getEnvId() != null) {
```
Comment (puthrayaharness) [inline 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java:1137]: What if the env has been made from some value to `null`

File: 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java [hunk 2]
```diff
@@ -1133,7 +1133,29 @@ public Workflow updateLinkedWorkflow(Workflow workflow, Workflow existingWorkflo
...
+      if (existingWorkflow.getEnvId() == null || !existingWorkflow.getEnvId().equals(workflow.getEnvId())) {
+        envChanged = true;
+      }
+    }
+    return updateWorkflow(workflow, workflow.getOrchestrationWorkflow(), envChanged, false);
+  }
+
+  @Override
+  public Workflow updateWorkflow(
+      Workflow workflow, OrchestrationWorkflow orchestrationWorkflow, boolean envChanged, boolean migration) {
```
Comment (puthrayaharness) [inline 400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java:1147]: envChanged is not being used here?

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by inderschera — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30741)* status: Some of the checks are too old. 2 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java, line 1137 at r1](https://reviewable.io/reviews/harness/harness-core/30741#-MvrtiEq_VSJhuWwo4fE-r1-1137:-Mw_SbHc5jxbH48T_TXc:bnpuiq4) ([raw file](https://github.com/harness/harness-core/blob/6745ee43911b7ad587425c3050bada96cd3c91b0/400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java#L1137)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

What if the env has been made from some value to `null`
</blockquote></details>

This would mean they have templatized environment in their workflow. We probably don't need to handle that case. We keep the already selected environment to auto populate environment when they switch back from templatized environment.

**Review by abhinavmittal-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30741)* status: Some of the checks are too old. 2 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java, line 1137 at r1](https://reviewable.io/reviews/harness/harness-core/30741#-MvrtiEq_VSJhuWwo4fE-r1-1137:-MwbKbnzC_hKhfwmOUIP:bppl4k3) ([raw file](https://github.com/harness/harness-core/blob/6745ee43911b7ad587425c3050bada96cd3c91b0/400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java#L1137)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

This would mean they have templatized environment in their workflow. We probably don't need to handle that case. We keep the already selected environment to auto populate environment when they switch back from templatized environment.
</blockquote></details>

Exactly what Inder said.
That is the behaviour via ui and we don't unset envId when we make it templatized
___
*[400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java, line 1147 at r1](https://reviewable.io/reviews/harness/harness-core/30741#-MvrtiEq_VSJhuWwo4fE-r1-1147:-Mw_diUZ2h5NqHaWgttG:bh8bs1d) ([raw file](https://github.com/harness/harness-core/blob/6745ee43911b7ad587425c3050bada96cd3c91b0/400-rest/src/main/java/software/wings/service/impl/workflow/WorkflowServiceImpl.java#L1147)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

envChanged is not being used here?
</blockquote></details>

yeah we don't use this.
This updateWorkflow implementation is not well written.

**Comment by abhinavmittal-harness** [comment]
trigger messagecheck

**Review by puthrayaharness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30741)* status: 2 file(s) to review.

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30741)* status: :shipit: complete! ✓

**Pull request merged**